portfolio/src/main/java/com/google/sps/servlets/GetCommentsServlet.java

```diff
@@ -0,0 +1,137 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet that stores and returns comments.
+ */
+@WebServlet("/get-comments")
+public class GetCommentsServlet extends HttpServlet {
+
+  // The max number of comments to send on a GET request. Is modified on POST request.
+  private int maxComments = 5;
+  private static final String COMMENT = "Comment";
+
+  /** 
+   * On GET request, writes to the response the comments list as a JSON string.
+   * @param request The request made by the connecting client.
+   * @param response The response that is sent back to the client.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Obtain and prepare comments from Datastore
+    Query commentsQuery = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery commentsPrepared = datastore.prepare(commentsQuery);
+
+    // Loop through each Comment entity until all comments are seen or until the max number of
+    // comments have been reached, and store in an ArrayList.
+    List<Comment> comments = new ArrayList<>();
+    Iterator<Entity> commentsIterator = commentsPrepared.asIterable().iterator();
+    int countComments = 0;
+    Entity entity;
+    while (commentsIterator.hasNext() && countComments < maxComments) {
+      entity = commentsIterator.next();
+
+      long id = entity.getKey().getId();
+      String email = (String) entity.getProperty("email");
+      String nickname = (String) entity.getProperty("nickname");
+      String text = (String) entity.getProperty("text");
+
+      // Determine if the logged in user is the owner of this comment;
+      boolean isOwner;
```

Comment:
```suggestion
      boolean isOwner = userService.isUserLoggedIn() && email.equals(userService.getCurrentUser().getEmail());
```

Reply:
Oh haha I don't know how I didn't see that I could have just done that, made the change.